src/colorizer/loaders/UrlArrayLoader.ts

```diff
@@ -7,29 +7,31 @@ import workerpool from "workerpool";
 import WorkerUrl from "./workers/urlLoadWorker?url&worker";
 
 import { FeatureArrayType, FeatureDataType } from "../types";
-import { packDataTexture } from "../utils/texture_utils";
+import { typedArrayToDataTexture } from "../utils/texture_utils";
 
 import { ArraySource, IArrayLoader } from "./ILoader";
 
 export class UrlArraySource<T extends FeatureDataType> implements ArraySource<T> {
   array: FeatureArrayType[T];
+  texture: DataTexture;
   type: T;
   min: number;
   max: number;
 
-  constructor(array: FeatureArrayType[T], type: T, min: number, max: number) {
+  constructor(array: FeatureArrayType[T], texture: DataTexture, type: T, min: number, max: number) {
     this.array = array;
     this.type = type;
     this.min = min;
     this.max = max;
+    this.texture = texture;
   }
 
   getBuffer(): FeatureArrayType[T] {
     return this.array;
   }
 
   getTexture(): DataTexture {
-    return packDataTexture(this.array, this.type);
```

Comment:
This step, `packDataTexture`, is now done once on the worker thread instead of whenever `getTexture()` is called.